src/BlocksPage.tsx

```diff
@@ -73,7 +75,11 @@ function BlockCard({ block, onShow }: BlockCardProps) {
       <Card.Body className="d-flex flex-column">
         <Card.Text>{blockTypeToLabel[block.type]}</Card.Text>
         <div className="mt-auto d-flex flex-column flex-md-row">
-          <Button variant="primary" onClick={onShow}>
+          <Button
+            variant="primary"
+            disabled={!astarte.token?.can('flow', 'GET', `/blocks/${block.name}`)}
+            onClick={onShow}
+          >
```

Comment:
This check should be run also to know whether to display a link in the card title or just a label for `block.name`